bigcache.go

```diff
@@ -41,11 +42,11 @@ const (
 )
 
 // NewBigCache initialize new instance of BigCache
-func NewBigCache(config Config) (*BigCache, error) {
```

Comment:
I think we should keep the old constructor for backward compatibility but mark it deprecated

Reply:
name suggestion for the new function `NewBigCacheWithContext()`? 

and what should we pass in the old function TODO() or Background() is fine?

Reply:
or just `New()` as the package name is bigcache

`bigcache.New()`

Reply:
Let's go with `WithContext` suffix as this is how std lib added context https://godocs.io/net/http#NewRequestWithContext

Reply:
Both background and todo returns the same empty context
```
var (
	background = new(emptyCtx)
	todo       = new(emptyCtx)
)
```
https://go.dev/src/context/context.go

Reply:
It's not about implementation, it's about readability 😉

Reply:
Saw the comment in `WithContext` now, went ahead with `New()`, looks more redable to me. Let me know if it requires a change.

Std lib follows both, see this https://pkg.go.dev/errors#New 
I think it depends on case by case basis. Also for `http.New()` wont really work as it is a multiple object inside http.